logger.js

```diff
@@ -61,13 +69,23 @@ function pinoLogger (opts, stream) {
   }
 
   function loggingMiddleware (req, res, next) {
+    var shouldLogSuccess = true
+
     req.id = genReqId(req)
     req.log = res.log = logger.child({req: req})
     res[startTime] = res[startTime] || Date.now()
-    if (!req.res) { req.res = res }
 
     if (autoLogging) {
-      res.on('finish', onResFinished)
+      if (req.url && autoLoggingIgnorePaths.length) {
+        for (let i = 0; shouldLogSuccess && i < autoLoggingIgnorePaths.length; i++) {
+          shouldLogSuccess = req.url.match(autoLoggingIgnorePaths[i]) == null
```

Comment:
This could be potentially be a large number of regular expression invocations. I would theorize that this could drastically slow down request processing.

Would a simple `autoLoggingIgnorePaths.includes(i)` be better?

Reply:
Yes, I can do that change.
I thought of regex to give more flexibility to the developer, but I guess that it would be dumb for someone to put "/" as it'd be just better to turn off the autologging. So it is developer responsibility in the end.

The correct would be 
```
shouldLogSuccess = req.url.includes(autoLoggingIgnorePaths[i]) == null
```

Reply:
This is one solution (using default Node URL class):
```
      if (req.url && autoLoggingIgnorePaths.length) {
        var url = new URL(req.url);
        shouldLogSuccess = !autoLoggingIgnorePaths.includes(url.pathname);
      }
```

Reply:
This is the other
```
      if (req.url && autoLoggingIgnorePaths.length) {
        for (let i = 0; shouldLogSuccess && i < autoLoggingIgnorePaths.length; i++) {
          shouldLogSuccess = req.url.includes(autoLoggingIgnorePaths[i]) == null
        }
      }
```

Reply:
However, if you put regex (which I agree is much slower), then we could at some point ignore:
`/myproduct/123/update`

with the two new solutions, this would not be possible.

I think it depends. Perhaps if we stay with simple urls, then we can see how it goes. If someone asks for more complexity, then we can add it later on.

Reply:
I went with the URL parsing solution. It solves the immediate need (health probes) and is fastest.

Reply:
Have you measured this? From my tests creating URL object was extremely slow.

Reply:
check https://github.com/Juntoz/test-url-parse
Indeed you were right. I replaced URL by fast-url-parser.